### ISSUE TITLE
WandbCallback lag [Q]

### ISSUE THREAD

**Issue by NourO93** [wandb/wandb#4269]
I am building a convolutional neural network using keras.compile, and during the training process I used WandbCallback to log the monitored loss into wandb. The problem is that there is significant delay in the logging process. It starts with 1-2 epochs behind and then gradually increases. The delay happens in both Google Colab and Spyder. I am unable to figure it out, so any tips would be appreciated. Please let me know what else I can provide to help make the issue clearer.

**Comment by ramit-wandb** [comment]
Hi @NourO93,

Are you logging a lot of data per epoch into W&B? That usually results in a slower upload. A link to your workspace would be great to understand what might be going on here.

Thanks,
Ramit

**Comment by NourO93** [comment]
> Hi @NourO93,
> 
> Are you logging a lot of data per epoch into W&B? That usually results in a slower upload. A link to your workspace would be great to understand what might be going on here.
> 
> Thanks, Ramit

You can view the code here:
[https://colab.research.google.com/drive/1TESYMceSxBgZ7Ji4uS31R4v7S_J3cZ3Q?usp=sharing](url)

**Comment by ramit-wandb** [comment]
Hi @NourO93,

I looked into the link you sent over, you are definitely not sending off too much data, but each epoch is also not very long (~10 seconds), that amount of delay is normal and expected in any project.

Thanks,
Ramit

**Comment by ramit-wandb** [comment]
​Hi @NourO93, We wanted to follow up with you regarding your support request as we have not heard back from you. Please let us know if we can be of further assistance or if your issue has been resolved.
​

**Comment by ramit-wandb** [comment]
​Hi @NourO93, since we have not heard back from you we are going to close this request. If you would like to re-open the conversation, please let us know!
